modules/library/coverage/src/main/java/org/geotools/coverage/processing/operation/BandSelector2D.java

```diff
@@ -233,12 +238,14 @@ static GridCoverage2D create(final ParameterValueGroup parameters, Hints hints)
         if (visibleBand == null) {
             visibleBand = visibleTargetBand;
         }
-        ParameterBlock params = new ParameterBlock().addSource(sourceImage);
+        // Using the ImageWorker 
+        ImageWorker w = new ImageWorker(sourceImage);
+        PlanarImage image;
         if (targetBands != sourceBands) {
-            operation = "BandSelect";
-            params = params.add(bandIndices);
+            image = w.setRenderingHints(hints).retainBands(bandIndices).getPlanarImage();
+        } else {
+            image = w.setRenderingHints(hints).nullOp().getPlanarImage();
```

Comment:
Interesting... if we have nothing to do, why call the null op? What does it do internally? The javadoc in ImageWorker does not really say why one would want to call it or why it's there to start with